docs/pages/reference/config.mdx

```diff
@@ -168,3 +168,51 @@ These settings apply to the Windows Desktop Service:
 (!docs/pages/includes/config-reference/desktop-config.yaml!)
 ```
 
+## Configuration versions
+
+In order to avoid breaking existing configurations, Teleport's configuration is
+versioned. The newer configuration version is `v3`. If a `version` is not
+specified in the configuration file, `v1` is assumed.
+
+Some new Teleport features require users to opt-in by explicitly upgrading their
+configuration to a newer version.
+
+### Config `v1`
+
+`v1` is the original version of Teleport's file configuration. It is still supported
+today, but most new users should start with the latest configuration version.
+
+### Config `v2`
+
+Configuration version `v2` was introduced in Teleport 8 as part of Teleport's
+TLS routing feature. With TLS routing, Teleport's proxy listens on a single port
+and uses ALPN and SNI to route incoming traffic to the correct listener rather
```

Comment:
nit:
```suggestion
and uses ALPN and SNI to route incoming traffic to the correct Teleport service rather
```
Probably for most people listener=port so maybe we can use use `service` word instead.

Reply:
It seems to me like listener or port would be correct. For example: when running a single Teleport service and connected to a database or openssh instance, the routing goes to a correct listener or port, not a correct Teleport service.